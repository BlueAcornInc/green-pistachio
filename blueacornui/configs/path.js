/**
 * Copyright Â© 2016 Magento. All rights reserved.
 * See COPYING.txt for license details.
 */

'use strict';

/**
 * Define paths.
 */

var webroot = '../webroot/';

module.exports = {
    binMage: webroot + 'bin/magento',
    pub: webroot + 'pub/static/',
    code: webroot + 'app/code/',
    design: webroot + 'app/design/',
    tmpLess: webroot + 'var/view_preprocessed/less/',
    tmpSource: webroot + 'var/view_preprocessed/source/',
    tmp: webroot + 'var',
    css: {
        setup: webroot + 'setup/pub/styles',
        updater: webroot + '../magento2-updater/pub/css'
    },
    less: {
        setup: webroot + 'app/design/adminhtml/Magento/backend/web/app/setup/styles/less',
        updater: webroot + 'app/design/adminhtml/Magento/backend/web/app/updater/styles/less'
    },
    uglify: {
        legacy: webroot + 'lib/web/legacy-build.min.js'
    },
    doc: webroot + 'lib/web/css/docs',
    spec: webroot + 'dev/tests/js/spec',
    static: {
        dir: webroot + 'dev/tests/static/testsuite/Magento/Test/Js/_files',
        whitelist: webroot + 'dev/tests/static/testsuite/Magento/Test/Js/_files/whitelist/',
        blacklist: webroot + 'dev/tests/static/testsuite/Magento/Test/Js/_files/blacklist/',
        tmp: webroot + 'validation-files.txt'
    },
    webroot: '../webroot/'
};
