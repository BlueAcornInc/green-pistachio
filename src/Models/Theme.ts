import glob from 'fast-glob';
import debug from 'debug';
const logger = debug('gpc:theme');

type ThemeConstructorArgs = ThemeData & {
    sourceDirectory: string;
    parent?: Theme;
};

// type CriticalCssInput = {
//     filepath: string;
//     urls: string[];
// };

export type ThemeData = {
    area: "frontend" | "adminhtml";
    path: string;
    enabled?: boolean;
    locales?: string[];
    stylesheets?: string[];
    // criticalCss?: CriticalCssInput[];
};

export default class Theme {
    private sourceDirectory: string;
    private parent?: Theme;
    private enabled: boolean | undefined;
    private data: ThemeData;

    constructor(config: ThemeConstructorArgs) {
        const {
            sourceDirectory,
            area,
            path,
            parent,
            locales,
            stylesheets,
            // criticalCss,
            enabled
        } = config;

        this.sourceDirectory = sourceDirectory;
        this.parent = parent;
        this.enabled = enabled || false;
        this.data = {
            area,
            path,
            locales,
            stylesheets,
            // criticalCss
        };
    }

    public configure(data: Partial<ThemeData>) {
        this.data = {
            ...this.data,
            ...data
        };
    }

    public getSourceDirectory() {
        return this.sourceDirectory;
    }

    public getParent() {
        return this.parent;
    }

    public getData() {
        return this.data;
    }

    public getLocales(): string[] {
        if (this.data.locales) {
            return this.data.locales;
        }

        logger(`${this.data.path}: Can't Determine Locales, defaulting to en_US`);

        return ['en_US'];
    }

    public getStyleSheets(): string[] {
        if (this.data.stylesheets) {
            return this.data.stylesheets;
        }

        logger(`${this.data.path}: Stylesheets not explicitly defined, detecting entries now.`);

        const files: Set<string> = new Set();

        const discoveredFiles = glob
            .sync(`${this.getSourceDirectory()}/**/[^_]*.less`)
            .map(sourceFile => sourceFile.replace(`${this.getSourceDirectory()}/web/`, ''));

        for (const file of discoveredFiles) {
            files.add(file);
        }

        let childTheme = this.getParent();

        while (childTheme) {
            for (const file of childTheme.getStyleSheets()) {
                files.add(file);
            }
            childTheme = childTheme.getParent();
        }

        const stylesheets = Array.from(files);

        logger(`${this.data.path}: Discovered the following stylesheets: ${stylesheets.join(', ')}`);

        return stylesheets;
    }

    // public getCriticalPaths(): CriticalCssInput[] {
    //     if (this.data.criticalCss) {
    //         return this.data.criticalCss;
    //     }

    //     logger(`No critical path data available for: ${this.data.path}`);

    //     return [];
    // }

    public getEnabled() {
        return this.enabled;
    }

    public setEnabled(enabled: undefined | boolean) {
        this.enabled = enabled;
    }
}
