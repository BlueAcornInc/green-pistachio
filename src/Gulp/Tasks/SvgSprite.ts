import { dest, parallel, src, TaskFunction, watch } from "gulp";
import glob from 'fast-glob';
import gulpPlumber from "gulp-plumber";
import svgSprite from "gulp-svg-sprite";
import debug from 'debug';
import { join, relative, dirname, basename } from 'path';
import Project from "../../Models/Project";
import Theme from "../../Models/Theme";
import { TaskInterface } from "./TaskInterface";
const logger = debug('gpc:gulp:svgSprite');

export default class SvgSprite implements TaskInterface {
    execute(project: Project): TaskFunction {
        const config = {
            svg: {
                percision: 4,
                xmlDeclaration: true,
                doctypeDeclaration: true,
                namespaceIDs: true,
                namespaceClassnames: true,
                dimensionAttributes: true
            },
            shape: {
                dest: '../src/intermediate-svg',
                spacing: {
                    padding: 0,
                    box: 'content'
                }
            },
            mode: {
                view: {
                    dest: '../css/',
                    prefix: '.svg-',
                    bust: false,
                    sprite: '../src/sprites.svg',
                    mixin: 'svg',
                    common: 'svg',
                    layout: 'vertical',
                    render: {
                        less: {
                            dest: '../css/source/blueacorn/_sprites.less',
                            template: join(__dirname, '../../../assets/tmpl/_sprite-mixins.less')
                        }
                    },
                    example: {
                        dest: '../../Magento_Theme/templates/framework/sprites.phtml',
                        template: join(__dirname, '../../../assets/tmpl/svg_sprites.phtml')
                    }
                }
            }
        };
        project.hooks.gulp.svgSpriteConfig.call(config);

        const tasks: TaskFunction[] = project.getThemes().map(theme => {
            const task: TaskFunction = done => {
                this.getSpriteFiles(theme).then(files => {
                    src(files, {
                        cwd: join(
                            theme.getSourceDirectory(),
                            'web',
                            'spritesrc'
                        ),
                        allowEmpty: true
                    })
                        .pipe(gulpPlumber())
                        .pipe(svgSprite(config).on('error', (err) => {
                            logger
                        }))
                        .pipe(dest(`${theme.getSourceDirectory()}/web/src/`))
                        .on('end', done);
                });
            };

            task.displayName = `svgSprite<${theme.getData().path}>`;

            return task;
        });

        return parallel(...tasks);
    }

    watch(project: Project): TaskFunction {
        return (done) => {
            watch(
                this.getSource(project.getThemes()),
                this.execute(project)
            );
        };
    }

    private getSource(themes: Theme[]): string[] {
        return themes.map(theme => `${theme.getSourceDirectory()}/web/spritesrc/**/*.svg`);
    }

    private async getSpriteFiles(targetTheme: Theme): Promise<string[] | string> {
        const svgMap: Map<string, string> = new Map();
        const targetPath = join(
            targetTheme.getSourceDirectory(),
            'web',
            'spritesrc'
        );

        let currentTheme: Theme | undefined = targetTheme;
        while (currentTheme) {
            const files = await glob(
                join(
                    currentTheme.getSourceDirectory(),
                    '/web/spritesrc/**/*.svg',
                )
            );

            for (const svgFile of files) {
                const normalizedFile = svgFile.replace(
                    join(
                        currentTheme.getSourceDirectory(),
                        'web',
                        'spritesrc'
                    ),
                    ''
                );
                const svgFileDir = dirname(svgFile);
                const svgFileName = basename(svgFile);
                const relativePath = relative(targetPath, svgFileDir);

                if (!svgMap.has(normalizedFile)) {
                    svgMap.set(normalizedFile, join(
                        relativePath,
                        svgFileName
                    ));
                }
            }

            currentTheme = currentTheme.getParent();
        }

        if (svgMap.size === 0) {
            logger(`No sprites found for: ${targetTheme.getData().path}`);
            return '**/*.svg';
        }

        return Array.from(svgMap.values());
    }
}
