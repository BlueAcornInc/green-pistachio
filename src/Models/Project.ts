import { join } from 'path';
import { SyncHook } from 'tapable';
import debug from 'debug';
import Module from "./Module";
import Theme, { ThemeData } from "./Theme";
import ConfigLoader from './Project/ConfigLoader';
import { Configuration } from 'webpack';
import WebpackConfigFactory from '../Webpack/ConfigFactory';
const logger = debug('gpc:project');

type ProjectConstructorArgs = {
    root: string;
    themes: Theme[];
    modules: Module[];
    enabledThemes?: string[];
};

type WebpackDevServerConfig = {
    proxyUrl: string;
    devServerUrl: string;
    devServerPort: number;
};

export default class Project {
    public root: string;
    private themes: Theme[];
    private enabledThemes: Theme[];
    private modules: Module[];
    private webpackDevelopmentMode: boolean = false;
    private proxyUrl: string = '';
    private devServerConfig: Partial<WebpackDevServerConfig> = {};

    public hooks = {
        // These hooks allow for modifying various gulp configurations
        gulp: {
            babelConfig: new SyncHook(["config"]),
            babelTypeScriptConfig: new SyncHook(["config"]),
            pngSpriteConfig: new SyncHook(["config"]),
            svgSpriteConfig: new SyncHook(["config"]),
            eslintConfig: new SyncHook(["config"]),
            criticalCssConfig: new SyncHook(["config"]),
        },
        webpack: {
            config: new SyncHook(["config"]),
            entryGlobs: new SyncHook(["globs"]),
            appendConfig: new SyncHook<[Configuration[], WebpackConfigFactory]>(["webpackConfigs", "configFactory"])
        },
        configure: new SyncHook(["project"])
    };

    public experiments = {
        webpack: {
            // Set to true in the the project.configure hook to enable unstable HMR functionality
            hmr: false,
            // Set to true to enable css modules
            cssModules: false,
            // Set to true to emit files to the pub directory instead of the individual theme directories
            emitFilesToPub: false,
        }
    };

    constructor(config: ProjectConstructorArgs) {
        const { root, themes, modules, enabledThemes } = config;

        this.root = root;
        this.themes = themes;
        this.modules = modules;
        this.enabledThemes = enabledThemes
            ? themes.filter(theme => enabledThemes.includes(theme.getData().path))
            : [];
    }

    public configure() {
        if (this.enabledThemes.length > 0) {
            // Enable default themes
            for (const theme of this.enabledThemes) {
                theme.setEnabled(true);
            }
        } else {
            for (const theme of this.themes) {
                if (
                    !theme
                        .getSourceDirectory()
                        .includes(
                            join(
                                this.getRootDirectory(),
                                'vendor'
                            )
                        )
                ) {
                    theme.setEnabled(true);
                    this.enabledThemes.push(theme);
                }
            }
        }

        // Enable default modules
        for (const magentoModule of this.modules) {
            if (
                !magentoModule
                    .getSourceDirectory()
                    .includes(
                        join(
                            this.getRootDirectory(),
                            'vendor'
                        )
                    )
            ) {
                magentoModule.setEnabled(true);
            }
        }

        this.hooks.configure.call(this);
    }

    public configureTheme(themeData: Partial<ThemeData>) {
        if (!themeData.path) {
            logger(`can not configure theme without defined path: ${JSON.stringify(themeData)}`);
            return;
        }

        const theme = this.themes.find(theme => theme.getData().path === themeData.path);

        if (!theme) {
            logger(`can not find theme with path = '${themeData.path}'`);
            return;
        }

        theme.configure(themeData);

        if (themeData.hasOwnProperty('enabled')) {
            theme.setEnabled(themeData.enabled);
        }
    }

    public enableModule(moduleName: string) {
        const magentoModule = this.modules.find(magentoModule => magentoModule.getName() === moduleName);

        if (!magentoModule) {
            logger(`could not find module by name: ${moduleName}`);
            return;
        }

        magentoModule.setEnabled(true);
    }

    public getThemes() {
        return this.enabledThemes.filter(magentoTheme => magentoTheme.getEnabled());
    }

    public getAllThemes() {
        return this.themes;
    }

    public getModules() {
        return this.modules.filter(magentoModule => magentoModule.getEnabled());
    }

    public getAllModules() {
        return this.modules;
    }

    public getRootDirectory() {
        return this.root;
    }

    public getThemePubDirectories(theme: Theme) {
        return theme.getLocales().map(locale => join(
            this.getStaticDirectory(),
            theme.getData().area,
            theme.getData().path,
            locale
        ));
    }

    public getWebpackOutputDirectories(theme: Theme) {
        if (this.isWebpackDevelopmentMode() || this.experiments.webpack.emitFilesToPub) {
            return theme.getLocales().map(locale => join(
                this.getRootDirectory(),
                ...(
                    this.isWebpackDevelopmentMode()
                    ? ['static']
                    : ['pub', 'static']
                ),
                theme.getData().area,
                theme.getData().path,
                locale
            ));
        }

        return [join(theme.getSourceDirectory(), 'web')];
    }

    public getStaticDirectory() {
        return join(
            this.getRootDirectory(),
            'pub',
            'static'
        );
    }

    public isWebpackDevelopmentMode() {
        return this.webpackDevelopmentMode;
    }

    public setWebpackDevelopmentMode() {
        this.webpackDevelopmentMode = true;
    }

    public setProxyUrl(url: string) {
        this.proxyUrl = url;
    }

    public getProxyUrl(): string {
        if (this.proxyUrl.length === 0) {
            if (!debug.enabled('gpc:logger')) {
                debug.enable('gpc:logger');
            }
            logger(`A project URL must be provided in your projects ${ConfigLoader.CONFIG_FILE}`);
            logger(`\n\nExample:
module.exports = project => {
project.hooks.configure.tap("Set Product Url", () => {
    project.setProxyUrl("https://example.test");
});
}
            `);

            throw new Error(`A project URL must be provided in your projects ${ConfigLoader.CONFIG_FILE}`);
        }

        return this.proxyUrl;
    }

    public setWebpackDevServerConfig(config: Partial<WebpackDevServerConfig>) {
        if (config.proxyUrl) {
            this.proxyUrl = config.proxyUrl;
        }

        this.devServerConfig = config;
    }

    public getWebpackDevServerConfig(): WebpackDevServerConfig {
        return {
            proxyUrl: this.getProxyUrl(),
            devServerUrl: 'green-pistachio.test',
            devServerPort: 8080,
            ...(this.devServerConfig || {})
        };
    }
}
