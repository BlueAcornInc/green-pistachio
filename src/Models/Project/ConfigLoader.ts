import { promises as fs } from 'fs';
import { join } from 'path';
import debug from 'debug';
import glob from 'fast-glob';
import Project from "../Project";
const logger = debug('gpc:configLoader');

export default class ConfigLoader {
    public static CONFIG_FILE = 'green-pistachio.config.js';

    constructor(private project: Project) {}

    get filePath(): string {
        return join(
            this.project.getRootDirectory(),
            ConfigLoader.CONFIG_FILE
        );
    }

    public async configExists(): Promise<boolean> {
        try {
            const configExists = await fs.stat(this.filePath);

            if (configExists) {
                return true;
            }
        } catch (err) {
            logger(`Config file can't be found at ${this.filePath}`);
        }

        return false;
    }

    public async loadConfig() {
        const configExists = await this.configExists();

        if (configExists) {
            try {
                this.interopRequire()(this.project);
            } catch (err) {
                logger(`Problem occurred while running user provided configuration file: ${err}`);
            }
        }

        await this.loadModularConfigFiles();
    }

    private async loadModularConfigFiles() {
        const filePaths = [
            ...this.project.getAllModules(),
            ...this.project.getAllThemes()
        ].map(compilableObject =>
            join(
                compilableObject.getSourceDirectory(),
                ConfigLoader.CONFIG_FILE
            )
        );

        const configFiles = await glob(filePaths);

        for (const configFile of configFiles) {
            try {
                this.interopRequire(configFile)(this.project);
            } catch (err) {
                logger(`Problem occurred while running user provided configuration file: ${err}`);
            }
        }
    }

    private interopRequire(filePath = '') {
        const configFile = filePath || this.filePath;
        const configModule = require(configFile);
        return configModule && configModule.__esModule ? configModule.default : configModule;
    }
}