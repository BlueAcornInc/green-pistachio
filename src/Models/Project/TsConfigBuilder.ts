import { promises as fs } from 'fs';
import { join } from 'path';
import debug from 'debug';
import Project from "../Project";
const logger = debug('gpc:typescript:configBuilder');

export default class TsConfigBuilder {
    public getFilePath(): string {
        return join(__dirname, '..', '..', '..', 'green-pistachio.tsconfig.json');
    }

    public async configFileExists(): Promise<boolean> {
        try {
            await fs.stat(this.getFilePath());

            return true;
        } catch (err) {
            return false;
        }
    }

    public async emitConfigFile(project: Project) {
        logger(`Generating Projects tsconfig.json base`);
        const include: string[] = [];
        const paths: Record<string, string[]> = {}

        for (const module of project.getModules()) {
            const moduleSrcDir = module.getSourceDirectory().replace(project.getRootDirectory(), '');

            const modulePaths: string[] = [
                join(moduleSrcDir, 'view/frontend/web/ts/*'),
                join(moduleSrcDir, 'view/base/web/ts/*'),
                join(moduleSrcDir, 'view/adminhtml/web/ts/*'),
            ];

            paths[`${module.getName()}/*`] = modulePaths.map(modulePath => modulePath.replace(/^\//, ''));

            include.push(join(moduleSrcDir, 'view/frontend/web/ts/**/*'));
            include.push(join(moduleSrcDir, 'view/base/web/ts/**/*'));
            include.push(join(moduleSrcDir, 'view/adminhtml/web/ts/**/*'));
        }

        for (const theme of project.getAllThemes()) {
            const themeSrcDir = theme.getSourceDirectory().replace(project.getRootDirectory(), '');

            include.push(join(themeSrcDir, '**/*'));
            
            // Add theme overrides to module paths
            for (const moduleName of Object.keys(paths)) {
                paths[moduleName].unshift(
                    join(themeSrcDir, moduleName.replace(/\/\*$/, ''), 'web/ts/*').replace(/^\//, '')
                );
            }
        }

        logger(`Resolved include paths to: ${JSON.stringify(include)}`);
        logger(`Resolved module paths to: ${JSON.stringify(paths)}`);

        const writePath = this.getFilePath();
        const config = this.getBaseConfig();

        config.include = include.map(includePath => `./${includePath}`);
        config.compilerOptions.paths = paths;

        try {
            logger(`Writing config file: ${writePath}`);
            await fs.writeFile(writePath, JSON.stringify(config, null, 4));
        } catch (err) {
            logger(`Problem writing config file: ${writePath}`);
            throw err;
        }

        return writePath;
    }

    private getBaseConfig() {
        return {
            "include": [
              "./app/**/*"
            ],
            "compilerOptions": {
              "baseUrl": ".",
              "paths": {},
              "strict": true,
              "esModuleInterop": true,
              "lib": [
                "dom",
                "es2015"
              ],
              "jsx": "react",
              "emitDecoratorMetadata": true,
              "experimentalDecorators": true,
              "target": "es5",
              "allowJs": true,
              "skipLibCheck": true,
              "allowSyntheticDefaultImports": true,
              "forceConsistentCasingInFileNames": true,
              "module": "esnext",
              "moduleResolution": "node",
              "resolveJsonModule": true,
              "isolatedModules": true,
              "noEmit": true
            }
          }
    }
}