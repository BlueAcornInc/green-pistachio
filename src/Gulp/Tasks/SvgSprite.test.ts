/**
 * @jest-environment node
 */
import Project from '../../Models/Project';
import Theme from '../../Models/Theme';
import { promises as fs } from 'fs';
import { join } from 'path';
import SvgSprite from './SvgSprite';

describe('Gulp: SVG Sprite', () => {
    it('should generate svg sprite', async (done) => {
        const root = join(
            __dirname,
            '__fixtures__',
            'svgsprite'
        );

        const expectedOutputFiles = [
            `${root}/vendor/magento/theme-frontend-blank/web/src/intermediate-svg/icon-stars.svg`,
            `${root}/vendor/magento/theme-frontend-blank/web/src/sprites.svg`,
            `${root}/vendor/magento/theme-frontend-blank/web/css/source/blueacorn/_sprites.less`,
            `${root}/vendor/magento/theme-frontend-blank/BlueAcorn_CmsFramework/templates/framework/sprites.phtml`,
        ];

        const project = new Project({
            themes: [
                new Theme({
                    sourceDirectory: `${root}/vendor/magento/theme-frontend-blank`,
                    area: 'frontend',
                    path: 'Magento/blank',
                    enabled: true
                })
            ],
            modules: [],
            root,
        });


        const svgSprite = new SvgSprite();
        svgSprite.execute(project)(async () => {
            for (const expectedOutputFile of expectedOutputFiles) {
                const contents = await fs.readFile(expectedOutputFile);
                expect(contents.toString()).toMatchSnapshot();
                await fs.unlink(expectedOutputFile)
            }

            done();
        });
    });
});
