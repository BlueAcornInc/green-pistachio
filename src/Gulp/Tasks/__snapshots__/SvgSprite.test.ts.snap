// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gulp: SVG Sprite should generate svg sprite 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><svg width=\\"121\\" height=\\"17\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 121 17\\"><g fill=\\"#003087\\" fill-rule=\\"evenodd\\"><path d=\\"M9 13.5l-5.29 2.781 1.01-5.89L.44 6.219l5.915-.86L9 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89zM35 13.5l-5.29 2.781 1.01-5.89-4.28-4.172 5.915-.86L35 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89zM60 13.5l-5.29 2.781 1.01-5.89-4.28-4.172 5.915-.86L60 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89zM86 13.5l-5.29 2.781 1.01-5.89-4.28-4.172 5.915-.86L86 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89zM112 13.5l-5.29 2.781 1.01-5.89-4.28-4.172 5.915-.86L112 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89z\\"/></g></svg>"`;

exports[`Gulp: SVG Sprite should generate svg sprite 2`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><svg width=\\"121\\" height=\\"17\\" viewBox=\\"0 0 121 17\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><view id=\\"icon-stars\\" viewBox=\\"0 0 121 17\\"/><svg width=\\"121\\" height=\\"17\\" viewBox=\\"0 0 121 17\\" id=\\"icon-stars-\\" xmlns=\\"http://www.w3.org/2000/svg\\"><g fill=\\"#003087\\" fill-rule=\\"evenodd\\"><path d=\\"M9 13.5l-5.29 2.781 1.01-5.89L.44 6.219l5.915-.86L9 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89zM35 13.5l-5.29 2.781 1.01-5.89-4.28-4.172 5.915-.86L35 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89zM60 13.5l-5.29 2.781 1.01-5.89-4.28-4.172 5.915-.86L60 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89zM86 13.5l-5.29 2.781 1.01-5.89-4.28-4.172 5.915-.86L86 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89zM112 13.5l-5.29 2.781 1.01-5.89-4.28-4.172 5.915-.86L112 0l2.645 5.36 5.915.859-4.28 4.172 1.01 5.89z\\"/></g></svg></svg>"`;

exports[`Gulp: SVG Sprite should generate svg sprite 3`] = `
"//
// @package     BlueAcorn/site
// @author      Blue Acorn iCi <code@blueacorn.com>
// @copyright   Copyright © Blue Acorn iCi. All rights reserved.
//

@sprite-url: \\"../images/sprites.svg\\";
@sprite-width: 121px;
@sprite-height: 17px;
@svg-icon-stars-x: 0;
@svg-icon-stars-y: 0;
@svg-icon-stars-xy: 0 0;
@svg-icon-stars-width: 121px;
@svg-icon-stars-height: 17px;

.svg() {
    background-image: url(\\"@{sprite-url}\\");
    background-repeat: no-repeat;
}

[class*=\\"svg-icon-\\"] {
    display: inline-block;
}

.svg-icon-stars() {
    background-position: @svg-icon-stars-xy;
}

.svg-icon-stars-dims() {
    width: @svg-icon-stars-width;
    height: @svg-icon-stars-height;
}

.svgi-icon-stars() {
    .svg();
    .svg-icon-stars();
    .svg-icon-stars-dims();
}

& when (@media-common = true) {
    .svg-icon-icon-stars {
        .svgi-icon-stars();
    }
}
"
`;

exports[`Gulp: SVG Sprite should generate svg sprite 4`] = `
"<?php
/**
 * @package     BlueAcorn/CmsFramework
 * @author      Blue Acorn iCi <code@blueacorn.com>
 * @copyright   Copyright © Blue Acorn iCi. All Rights Reserved.
 */

// @codingStandardsIgnoreFile
// phpcs:disable Generic.Files.LineLength.TooLong

/**
 * CMS Framework Documentation Template
 *
 * @var $block \\\\BlueAcorn\\\\CmsFramework\\\\Block\\\\CmsFramework
 */
?>

<div class=\\"ba-grid margin-bottom-40 padding-left-0 padding-right-0\\">
    <h3 class=\\"item span-12\\">Sprites</h3>

    <h4 class=\\"item span-12\\">icon-stars</h4>

    <div class=\\"item span-s-12 d-flex flex-wrap margin-bottom-5\\">
        <div class=\\"padding-10 float-left d-flex justify-content-center align-content-center bg-white c-contrast-white\\">
            <i class=\\"svg-icon-icon-stars\\"><span class=\\"visually-hidden\\">icon-stars</span></i>
        </div>
        <div class=\\"padding-10 float-left d-flex justify-content-center align-content-center bg-black c-contrast-black\\">
            <i class=\\"svg-icon-icon-stars\\"><span class=\\"visually-hidden\\">icon-stars</span></i>
        </div>
        <dl class=\\"padding-10 float-left bg-white c-contrast-white\\">
            <dt>Width</dt>
            <dd>121</dd>
            <dt>Height</dt>
            <dd>17</dd>
            <dt>Background Position Mixin</dt>
            <dd>.svg-icon-stars()</dd>
            <dt>Dimensions Mixin</dt>
            <dd>.svg-icon-stars-dims()</dd>
            <dt>Full Mixin</dt>
            <dd>.svgi-icon-stars()</dd>
            <dt>Class</dt>
            <dd>.svg-icon-icon-stars()</dd>
        </dl>
    </div>

    <pre class=\\"item span-s-12 margin-bottom-20\\"><code>&lt;i class=\\"svg-icon-icon-stars\\"&gt;&lt;span class=\\"visually-hidden\\"&gt;icon-stars&lt;/span&gt;&lt;/i&gt;</code></pre>
</div>
"
`;
